### PR TITLE
[staging] python38Packages.cffi: fix tests

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
follow up to: #108430

returning `None` is expected behavior when calling `ctypes.utils.find_library`. Disable tests which assume otherwise.

cc @FRidh 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg build python3Packages.pandas

**Comment by jonringer** [comment]
I was able to build `python3Packages.pandas` locally
```
[16:26:29] jon@nixos ~/projects/nixpkgs (fix-py38-cffi)
$ nix build -f . python3Packages.pandas
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy
[13 built]
```

**Comment by peterhoeg** [comment]
Without this python2Packages.cffi and therefore nixops are both broken.

**Comment by FRidh** [comment]
> Without this python2Packages.cffi and therefore nixops are both broken.

https://github.com/NixOS/nixpkgs/pull/108569

File: pkgs/development/python-modules/cffi/default.nix [hunk 1]
```diff
@@ -33,12 +33,32 @@ if isPyPy then null else buildPythonPackage rec {
     "-Wno-unused-command-line-argument -Wno-unreachable-code";
 
   doCheck = !stdenv.hostPlatform.isMusl && !stdenv.isDarwin; # TODO: Investigate
-  checkPhase = ''
-    py.test -k "not test_char_pointer_conversion"
-  '';
+  disabledTests = [
+    # unix gets mistaken as windows because find_library returns None
```
Comment (orivej) [inline pkgs/development/python-modules/cffi/default.nix:37]: The test diagnostics could be better, but the fact that `find_library('c')` returns None and the test falls through to the Windows case indicates that find_library is broken.

**Review by orivej — CHANGES_REQUESTED** [review]
python38Packages.cffi tests became broken on staging when #108430 (commited to staging) met with e1ed9b3b0851b95d3e82fb1085a121531c32a42a (commited to staging-next) in 2f7fa4bb8887cc90f72a2b9173b603c9af9d62ea. The right fix is to restore the patch for 3.8.6 and remove when updating to 3.8.7.

**Comment by jonringer** [comment]
alternate pr: #108601

**Comment by FRidh** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/108601.

**Pull request closed without merging**